### PR TITLE
timezone: Add support for macOS

### PR DESCRIPTION
**Pull request by indrajitr**
##### SUMMARY

Add support for macOS in timezone module.

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

timezone

##### ANSIBLE VERSION

```
ansible 2.4.0 (devel 753d30b64d) last updated 2017/04/10 01:06:44 (GMT -500)
  config file = 
  configured module search path = [u'/Users/indrajit/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/indrajit/Projects/Community/Ansible/gitrepo/ansible/lib/ansible
  executable location = /Users/indrajit/Projects/Community/Ansible/gitrepo/ansible/bin/ansible
  python version = 2.7.13 (default, Feb 13 2017, 23:32:07) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)]

```

##### ADDITIONAL INFORMATION

On macOS, preferred way of managing timezone is via `systemsetup(8)`. Thus, we
use this command instead of relying on directly modifying `/etc/localtime` as in
other *BSDs.

### CODE REVIEW

File: lib/ansible/modules/system/timezone.py
Comment (tmshn) [inline lib/ansible/modules/system/timezone.py]: Use `%` instead of `.format()` to support older version of Python.
```python
self.abort('%s is not a supported option on target platform' % key)
```

(L557 too)

**Review by tmshn — CHANGES_REQUESTED** [review]
Thanks for nice work!
I just commented one nitpicky point.

**Comment by tmshn** [comment]
And add your name here! 📇 
https://github.com/indrajitr/ansible/blob/d41a835c8fcfcf7120411232dc8c83bedaa16b19/lib/ansible/modules/system/timezone.py#L61

File: lib/ansible/modules/system/timezone.py
Comment (indrajitr) [reply lib/ansible/modules/system/timezone.py]: Agreed! I kept it for the sake of uniformity with other `self.module.fail_json()` calls in the module.
So how about changing all the occurrences of `.format()` with `%` across all the classes in the module?

Comment (tmshn) [reply lib/ansible/modules/system/timezone.py]: Keep using `self.module.fail_json()`, of course fine. 👍 
Changing all `.format()` into `%`, great! 👏 

I've never noticed that the module has so many `.format()`...my reviews must be too rough at that time 😓

**Comment by indrajitr** [comment]
@tmshn, updated. Thanks for the feedback!

**Comment by tmshn** [comment]
What about the L504?
https://github.com/indrajitr/ansible/blob/25f4e472d3d6d4bcab3d54a6b5b036324e142e17/lib/ansible/modules/system/timezone.py#L504

This is in the `SmartOSTimezone`, which is out of the scope of this PR, but I hope it to be changed together...

**Comment by indrajitr** [comment]
@tmshn, sorry missed that one. Corrected now :)

**Review by tmshn — APPROVED** [review]
Thank you for your quick fixes. Now looks great 👍 

Say "ready_for_review" to prompt other maintainers to review.
https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md

**Comment by indrajitr** [comment]
ready_for_review

**Comment by tmshn** [comment]
shipit

**Comment by indrajitr** [comment]
bot_status

**Comment by indrajitr** [comment]
shipit

**Comment by tmshn** [comment]
@jasperla It seems what we are waiting for is your review. Could you review this?

**Comment by mscherer** [comment]
bot_status

**Comment by mscherer** [comment]
shipit

**Comment by tmshn** [comment]
Oh it loses my `ship it` comment...

**Comment by tmshn** [comment]
shipit

**Pull request merged**
